src/main/java/teammates/ui/webapi/CreateCourseAction.java

```diff
@@ -83,10 +84,14 @@ public JsonResult execute() throws InvalidHttpRequestBodyException, InvalidOpera
             logicNew.createCourseAndInstructor(userInfo.getId(), sqlCourseAttributes);
 
         } catch (EntityAlreadyExistsException e) {
+            HibernateUtil.getSessionFactory().getCurrentSession().getTransaction().rollback();
```

Comment:
If we are able to handle rollbacks in `doFilter`, then the code here and `CreateInstructorAction.java` can be removed, and we won't have to keep adding rollback code down the line.